### PR TITLE
YETUS-1102. Add an option to comment to GitHub PR.

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/YETUS-1102

- Add `--github-write-comment` to back to the previous behavior.
- Add back `--github-use-emoji-vote` option.

### CODE REVIEW

File: precommit/src/main/shell/plugins.d/github.sh [hunk 1]
```diff
@@ -1055,6 +1059,11 @@ function github_finalreport
     return 0
   fi
 
+  if [[ "${GITHUB_WRITE_COMMENT}" == true ]]; then
+    github_finalreport_as_comment result
+    return 0
```
Comment (aw-was-here) [inline precommit/src/main/shell/plugins.d/github.sh:1064]: You cannot do a return here as it cuts out the rest of the code.

**Review by aw-was-here — CHANGES_REQUESTED** [review]

**Comment by aajisaka** [comment]
I'll look into this.
```
|  -1  |      docker  |   6m 27s   | Docker failed to build 
|      |              |            | yetus/yetus:b5a09398.
```

**Comment by aajisaka** [comment]
Re-ran all the job, and linkcheck is still failing:
```
  #44 ERROR: lstat /var/lib/docker/tmp/buildkit-mount087864928/root/go/bin: no such file or directory
  ------
   > importing cache manifest from apache/yetus-base:pull_213_merge:
  ------
  ------
   > importing cache manifest from apache/yetus-build:pull_213_merge:
  ------
  ------
   > importing cache manifest from apache/yetus:pull_213_merge:
  ------
  ------
   > [stage-10 24/25] COPY --from=checkmake /root/go/bin/* /usr/local/bin:
  ------
  lstat /var/lib/docker/tmp/buildkit-mount087864928/root/go/bin: no such file or directory
  Error: Process completed with exit code 1.
```
Probably the test referenced the old build cache.

**Comment by aajisaka** [comment]
GitHub Actions doesn't have "clear cache" feature. We have to change the cache key to clear the cache, and it is very frustrating.
https://github.com/actions/cache/issues/2

**Comment by aajisaka** [comment]
After digging this issue, this issue is not related to the cache of GitHub Actions. However, I'm not sure why the old Dockerfile is used.
https://pipelines.actions.githubusercontent.com/4IRN7cmKQG30mFwpZCDL2WwRpIOhhBRpKKjsnasVdYJ4dd7vJI/_apis/pipelines/1/runs/793/signedlogcontent/3?urlExpires=2021-03-04T07%3A54%3A16.4608893Z&urlSigningMethod=HMACV1&urlSignature=B%2FSsQekc2fFD%2FI5O9vUFrVgS2ZMYrRwlqmcCw5kCjck%3D
```
2021-03-04T07:47:56.7163921Z #43 [checkmake 1/1] RUN go get -u github.com/mrtazz/checkmake
2021-03-04T07:47:56.7164901Z #43 sha256:a79f93a32d6630e29c4cb62f8fca54b97bebb0c0135d1e4b6e0bc81fca55b104
2021-03-04T07:47:56.7166093Z #43 2.648 go: downloading github.com/mrtazz/checkmake v0.0.0-20201029101907-575315c9924d
2021-03-04T07:47:56.7166602Z #43 DONE 4.0s
```

File: precommit/src/main/shell/plugins.d/github.sh (hunk 1 above)
Comment (aajisaka) [reply precommit/src/main/shell/plugins.d/github.sh]: Fixed https://github.com/apache/yetus/pull/213/commits/18bde22af1134f42ba0afa87755b72f9516ebfbd

**Comment by aajisaka** [comment]
I tested in https://github.com/apache/hadoop/pull/2745, and the comment looks good. I'll commit this tomorrow if there are no objections.

**Comment by aajisaka** [comment]
72 hours passed since the last commit. Merging this.

**Pull request merged**